ml-agents/mlagents/trainers/trainer/rl_trainer.py

```diff
@@ -142,12 +143,17 @@ def advance(self) -> None:
                 # We grab at most the maximum length of the queue.
                 # This ensures that even if the queue is being filled faster than it is
                 # being emptied, the trajectories in the queue are on-policy.
+                _queried = False
                 for _ in range(traj_queue.qsize()):
+                    _queried = True
                     try:
                         t = traj_queue.get_nowait()
                         self._process_trajectory(t)
                     except AgentManagerQueue.Empty:
                         break
+                if self.threaded and not _queried:
+                    # Avoid busy-waiting
+                    time.sleep(0.0001)
```

Comment:
@harperj the alternative solution here might be to use a blocking get with timeout when not `_queried`. (similar to old solution with `empty_queue`). I've tested both and it seems the performance is similar.

Reply:
If this works to yield I think it's fine.  You could also make this `time.sleep(0)` I believe.

Reply:
IIRC time.sleep(0) only works on some platforms, annoyingly.